bin/aplimits

```diff
@@ -0,0 +1,381 @@
+        of background rate is not desired.
+        Units: sec        
+    max_counts : integer
+        Max total counts before switching from Baysian approach to assuming
+        that the background rate is known exactly, even if the input is given
+        as background counts and aperture area.
+
+        If the number of background counts is large, the Baysian posterior
+        becomes sharply peaked. In his case the algorithms for the numerical
+        integration may run very long or even fail while result is only
+        negligibly different from those obtained assuming that the background
+        is known exactly.
+
+        Comparison of both methods indicates that a value of max_counts=50
+        (the default) is a good choice for a wide range of input parameters.
+            
+    Returns
+    -------
+    lambs : float
+        Upper limit for the source intensity (in cts /area / time)
+        
+    Note
+    ----
+    The probabilities for Type I and Type II errors are called alpha and beta
+    in the Kashyap et al. (2010) paper but the same symbols are used with a
+    different meaning in aprates. To avoid confusion, the parameters of this
+    function are named typeIerror and typeIIerror.
+    '''
+    if typeIerror <=0 or typeIerror >= 1:
+        raise ValueError('typeIerror is a probability, which must be between 0 and 1.')
+    if typeIIerror <=0 or typeIIerror >= 1:
+        raise ValueError('typeIIerror is a probability, which must be between 0 and 1.')
+    if bkg_rate is None and m is None:
+        raise ValueError('Either bkg_rate or m have to be given.')
+    if bkg_rate is not None and m is not None:
+        v1('bkg_rate and m are given. Using bkg_rate for the computation.')
+    if bkg_rate is None and m >= max_counts:
+        bkg_rate = m / A_b / T_b * A_s * T_s
+    if bkg_rate is not None:
+        out = APLimits(lamb=bkg_rate)(typeIerror, typeIIerror)
+    else:
+        limitfinder = APLimitsMarginalize(nb=m, taus=T_s, taub=T_b, r=A_b/A_s)
+        out = limitfinder(typeIerror, typeIIerror)
+    return out
+#
+# Main Routine
+#
+@lw.handle_ciao_errors(TOOLNAME, __REVISION__)
+def main():
+    'Main routine'
+    from ciao_contrib.param_soaker import get_params
+    import paramio as pio
+
+    # get parameters
+    pars = get_params(TOOLNAME, "rw", sys.argv,
+                      verbose={"set": lw.set_verbosity, "cmd": v1})
+    ul, sstar = aplimits(typeIerror=float(pars["typeIError"]),
+                         typeIIerror=float(pars["typeIIerror"]),
+                         T_s=float(pars["T_s"]),
+                         A_s=float(pars["A_s"]),
+                         bkg_rate=float(pars["bkg_rate"])
+                         m=int(pars["m"]),
+                         T_b=float(pars["T_s"]),
+                         A_b=float(pars["A_s"]),
+                         max_counts=int(pars["max_counts"])
+
+    ph = pio.paramopen(TOOLNAME + "_out")
```

Comment:
There are 2 options

1. Either store the values back into the tools own parameter file, like `dmcoords` does or
2. Store the values into a new parameter file like `aprates`

For the `aprates` approach, the paramio module isn't helpful -- it can only be used to set/get values it can't create parameter files from scratch.  So you'd probably want to add an `outfile` parameter and then just something like 

```python
with open(outfile, "rw") as fp:
    fp.write(f's_star,r,h,{sstar},,,"Some comment"\n"')
    fp.write(f'upper_limit,r,h,{ul},,,"Upper limit"\n')
```

You might want to consider saving the PDF as a table (and then these scalar values could be written as keywords).  `aprates` doesn't, but it does print the PDF values when the verbose is >=1.  I don't know of a specific use for it, but it's something Aneta and Frank had asked about at different points.

Reply:
I think that `aprates` approach is more appropriate here. In `dmcoords` all coordinate representations are interchangable, i.e. I can put in X,Y to calculate RA/DEC, or I can put in RA/DEC to calculate X,Y. Here, it's more like `aprates` with input and output parameters.